template/template.go

```diff
@@ -49,140 +41,33 @@ func FromGlobs(paths ...string) (*Template, error) {
 		}
 	}
 
-	t.funcs(DefaultFuncs)
 	return t, nil
 }
 
-type FuncMap map[string]interface{}
-
-func (t *Template) funcs(fm FuncMap) *Template {
-	t.text.Funcs(text_tmpl.FuncMap(fm))
-	t.html.Funcs(html_tmpl.FuncMap(fm))
-	return t
-}
-
-func (t *Template) ExecuteTextString(name string, data interface{}) (string, error) {
+func (t *Template) ExecuteTextString(text string, data interface{}) (string, error) {
+	tmpl, err := t.text.Clone()
+	if err != nil {
+		return "", err
+	}
+	tmpl, err = tmpl.New("").Parse(text)
+	if err != nil {
+		return "", err
+	}
 	var buf bytes.Buffer
-	err := t.ExecuteText(&buf, name, data)
+	err = tmpl.Execute(&buf, data)
 	return buf.String(), err
 }
 
-func (t *Template) ExecuteText(w io.Writer, name string, data interface{}) error {
-	return t.text.ExecuteTemplate(w, name, data)
-}
-
-func (t *Template) ExecuteHTMLString(name string, data interface{}) (string, error) {
+func (t *Template) ExecuteHTMLString(html string, data interface{}) (string, error) {
```

Comment:
Compiling the templates on each execution is an anti-pattern in my opinion. Even though we have it in Prometheus too. Templates are compilable for a reason.

Not feeling very strongly, but I also see it as an advantage to avoid races and as a result broken notifications.
I don't see much benefit to be constantly able to change the templates – it happens to rarely that sending a `SIGHUP` is not too much overhead.